### PR TITLE
Refs #12489 - enable sudo

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

**Comment by iNecas** [comment]
The proxy now allows to use sudo to switch to the effective user from the remote one

**Comment by stbenjam** [comment]
Looks good, haven't tried it yet.

How will the Foreman side decide to use sudo or not?

**Comment by iNecas** [comment]
Similar to the mechanism we have son the ssh_user - using a host param, default to settings

**Comment by iNecas** [comment]
The PR on foreman_remote_execution is finally here https://github.com/theforeman/foreman_remote_execution/issues/80

File: lib/smart_proxy_remote_execution_ssh/dispatcher.rb [hunk 1]
```diff
@@ -7,18 +7,19 @@ module Proxy::RemoteExecution::Ssh
...
-        @ssh_user         = data[:ssh_user]
-        @effective_user   = data[:effective_user]
-        @script           = data[:script]
-        @host_public_key  = data[:host_public_key]
-        @suspended_action = data[:suspended_action]
+        @id                    = data[:id]
+        @host                  = data[:host]
+        @ssh_user              = data[:ssh_user]
+        @effective_user        = data[:effective_user]
+        @effective_user_method = data[:effective_user_method] || 'su'
```
Comment (stbenjam) [inline lib/smart_proxy_remote_execution_ssh/dispatcher.rb]: Can we make the default 'sudo' to match Foreman's default? Or was there a reason for picking su?

Comment (stbenjam) [inline lib/smart_proxy_remote_execution_ssh/dispatcher.rb]: Nevermind, I see, we used to use `su` so this keeps the behavior of the proxy the same with older Foremans.

**Comment by stbenjam** [comment]
Works well, ACK from me

**Comment by iNecas** [comment]
Thanks @stbenjam for review

**Pull request merged**